lib/hummingbird_survey/models/survey.rb

```diff
@@ -20,13 +20,17 @@ def surveyed_data_for(surveyed_obj)
       end
     end
 
-    def set_surveyed_data_for!(surveyed_obj, target_data)
+    def set_surveyed_data_for!(surveyed_obj, target_data, request = nil)
       if sub_obj.present?
         surveyed_obj.data["survey_#{self.id}_#{sub_obj.class.name}_#{sub_obj.id}"] = target_data
       else
         surveyed_obj.data["survey_#{self.id}"] = target_data
       end
 
+      if request.present? && request.is_a?(ActionDispatch::Request)
+        surveyed_obj.data["survey_#{self.id}_request"] = { ip_address: request.remote_ip, browser: request.user_agent, path: request.path }
```

Comment:
@cgratigny : why not make this part of the hash for the survey? `surveyed_obj.data["survey_#{self.id}"]["request"]`